region/src/main/java/org/motechproject/nms/region/service/impl/BaseLocationImportService.java

```diff
@@ -13,17 +13,33 @@
 import javax.validation.ConstraintViolationException;
 import java.io.IOException;
 import java.io.Reader;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
 public abstract class BaseLocationImportService<T> {
 
+    public static final String PARENT_STATE = "state";
+    public static final String PARENT_DISTRICT = "district";
+    public static final String PARENT_TALUKA = "taluka";
+    public static final String PARENT_HEALTH_BLOCK = "healthBlock";
+
     private Class<T> type;
     private MotechDataService<T> dataService;
+    private Map<String, Object> parents;
```

Comment:
Placing a property in a service makes the service stateful. We should avoid it, as it makes it prone to concurrency issues. Consider passing the map as a parameter to the importData(..) method.